BitmarkRegistry/Services/AssetService.swift

```diff
@@ -15,11 +15,21 @@ class AssetService {
     return FileUtil.computeFingerprint(data: data)
   }
 
-  static func getAsset(from fingerprint: String) -> Asset? {
+  static func getAsset(from fingerprint: String) -> AssetR? {
     guard let assetId = computeAssetId(fingerprint: fingerprint) else { return nil }
 
     do {
-      return try Asset.get(assetID: assetId)
+      let userRealm = try RealmConfig.currentRealm()
+      if let assetR = userRealm?.object(ofType: AssetR.self, forPrimaryKey: assetId) {
+        return assetR
+      }
+    } catch {
+      ErrorReporting.report(error: error)
+    }
+
+    do {
```

Comment:
why don't we combine into one try/catch?

Reply:
because if we put it into one try/catch: if there are an error while getting the local asset; we still have to get the asset from server; and that will require another try/catch before really return to nil.

Reply:
Ok.